### DISCUSSION TITLE
Báo lỗi đã ghi nhận - Cập nhật 2023/05/12

### DISCUSSION THREAD

**Discussion by thinh-vu** [thinh-vu/vnstock#16] [Chia sẻ thành tích (Show and Tell)]
Tất cả báo lỗi được ghi nhận qua thử nghiệm trên môi trường tiêu chuẩn là Google Colab.
---
1. Hàm `listing_companies()` không thể tải dữ liệu trong môi trường Google Colab, hoạt động bình thường với macOS, Windows
2. Hàm `stock_intraday_data()` không hoạt động, có thể do cập nhật key của JSON data từ API request.
3. Hàm `market_top_mover()` và `fr_trade_heatmap()` dùng để thu thập dữ liệu toàn cảnh thị trường, nguyên nhân sơ bộ qua báo lỗi là do bị chặn từ phía nguồn cấp dữ liệu (SSI, FiinTrade).
---
Đây là một số lưu ý dành cho các bạn khi gặp và báo lỗi với vnstock:

- Ưu tiên sử dụng môi trường Google Colab để thử nghiệm bởi đây là môi trường tiêu chuẩn, dễ debug và sửa lỗi.
- Nếu các bạn sử dụng vnstock cho mục đích chạy các workflow tự động, sử dụng terminal/máy chủ cloud thì khi gặp lỗi, vui lòng gửi thông tin chi tiết về môi trường thực thi bao gồm: 
   - Hệ điều hành: Linux, macOS, Windows
   - Phiên bản hệ điều hành: Windows 10, macOS Ventura, Ubuntu 23.4, vv. Bạn có thể chụp thông tin bản dựng của máy đang sử dụng để đính kèm. Việc này giúp dễ xác định được lỗi cục bộ hay lỗi nghiêm trọng với toàn bộ thư viện.
- Hình ảnh báo lỗi, hoặc có thể copy toàn bộ thông báo, lưu vào 1 file text và chia sẻ để tôi có thêm thông tin chi tiết về lỗi.

![image](https://github.com/thinh-vu/vnstock/assets/39723360/afca1f98-4a82-4b56-a4c5-afd3f90644e0)

Thịnh sẽ dành thời gian xem lại và cập nhật bản sửa lỗi trong những ngày tới. Cám ơn các bạn đã quan tâm <3

**Comment by HuyQuocTran2006** [comment]
hàm listing_companies chắc làm tay luôn đi Thịnh. tạo 1 cái dataframe có đầy đủ 1619 mã cổ phiếu

**Reply by thinh-vu** [reply]
@HuyQuocTran2006 Hàm này các bạn chỉ cần chạy ở máy local là được, có thể bên SSI, FiinTrade họ chặn theo domain hoặc toàn bộ truy cập từ Linux OS (cloud service).
Đặc thù của danh sách các công ty niêm yết là nó không thay đổi nhiều. Do đó các bạn chỉ cần save dataframe có được từ hàm vào file csv 1 lần và lâu lâu kiểm tra lại để lấy dữ liệu (nếu có các workflow tự động). Screenshot này mình chạy trên terminal của Windows.
![image](https://github.com/thinh-vu/vnstock/assets/39723360/e2a457fc-d3b9-480f-ba99-81ea2fed05e2)

**Comment by thinh-vu** [comment]
Đã xử lý lỗi của issue 1 và 2. Thay đổi được update vào `beta` branch.
- Changes log: https://github.com/thinh-vu/vnstock/blob/beta/changes_log.md
- Readme: Cập nhật mô tả cho 2 hàm đã thay đổi, cần cài đặt package ở beta branch cho project của các bạn để nhận cập nhật sửa lỗi

Lỗi 3 do SSI bổ sung lá chắn bảo mật để chặn bot trước khi chuyển hướng API để truyền dữ liệu. Sẽ cần thêm thời gian để xử lý.

**Comment by Yunkaidasick** [comment]
Lỗi khi muốn khởi tạo đối tượng mới là một mã cổ phiếu, nhưng đối tượng được tạo lại là VN30F1M
 
![image](https://github.com/thinh-vu/vnstock/assets/168903676/e5eed4d3-ce0a-4b5c-8ea8-00bde1c9ba99)

**Reply by thinh-vu** [reply]
Chào @Yunkaidasick 
Cám ơn bạn đã báo lỗi. Mình đã kiểm tra và ghi nhận lỗi này. Trong thời gian chờ nâng cấp, bạn sử dụng tham số `symbol='MBB'` trong hàm history để áp dụng thay đổi mã CK chính xác nhé. Hiện tại hành vi của hàm này là ghi đè giá trị của symbol được cập nhật gần nhất (tức ở hàm history) bằng giá trị mặc định VN30F1M.

**Reply by thinh-vu** [reply]
Hiện tại mình đã khắc phục xong lỗi nói trên và cập nhật phiên bản 0.3.0.2. Bạn có thể theo dõi chi tiết cập nhật [tại đây](https://vnstocks.com/docs/tai-lieu/lich-su-phien-ban)
